viewing_party/party.py

```diff
@@ -1,23 +1,248 @@
+        most_watched_genre = max(genre_count, key = genre_count.get)
+        return most_watched_genre
+    else:
+        return None
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    #takes in user_data (a dictionary with 3 keys, watched and watchlist and friends)
+    #now, user_data has a third key: "friends".
+    #the value associated with this is a list of dictionaries, one dictionary per friend.
+    # The dictionary for each friend has "watched", with a list of movies. (which is itself a dict, remember.) 
+    #This function outputs the list of movies (dicts) in watched that are unique, and not watched by another friend.
+    
+    user_watched_titles = get_user_watched_titles(user_data)
+    friend_watched_titles = get_friends_watched_titles(user_data)
+
+    #set difference
+    unique_watched_titles = user_watched_titles - friend_watched_titles
+
+    #retrieve the dictionary entries from the original user_data
+    unique_watched_movies = []
+    for movie in user_data["watched"]:
+        if movie["title"] in unique_watched_titles:
+            unique_watched_movies.append(movie)
+            unique_watched_titles.remove(movie["title"]) #remove it to prevent duplicates
+    return unique_watched_movies
+
+def get_user_watched_titles(user_data):
+    #returns a set of titles user has watched.
+    user_watched_titles = []
+    for movie in user_data['watched']:
+        user_watched_titles.append(movie["title"])
+    return set(user_watched_titles)
+
+def get_friends_watched_titles(user_data):
+    #returns a set of titles of movies friend has watched.
+    friend_watched_titles = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_watched_titles.append(movie["title"])
+    return set(friend_watched_titles)
+
+def get_favorite_watched_titles(user_data):
+    #returns a set of titles user has watched.
+    user_watched_titles = []
+    for movie in user_data['favorites']:
+        user_watched_titles.append(movie["title"])
+    return set(user_watched_titles)
+
+
+def get_friends_unique_watched(user_data):
+    #this takes in user_data (with keys including 'watched' and 'friends')
+    # it returns a list of dictionaries of all the movies a friend(s) has watched that user hasn't.  
+    friend_watched_titles = get_friends_watched_titles(user_data)
+    user_watched_titles = get_user_watched_titles(user_data)
+    friends_unique_titles = friend_watched_titles - user_watched_titles
+
+    #retrieve info from user_data matching the titles
+    unique_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] in friends_unique_titles and movie not in unique_watched:
+                unique_watched.append(movie)

```

Comment:
Nice! Helper functions made this implementation logic efficient and well organized.